backend/app/Services/FileParser.php

```diff
@@ -2,31 +2,186 @@
+     * otherwise create a new item.
+     *
+     * @param $files
+     */
+    public function store($files)
     {
-      $item = $this->item->where('title', $title)->orWhere('original_title', $title)->first(['id', 'tmdb_id']);
+      foreach($files as $type => $items) {
+        foreach($items as $item) {
+          $title = $this->isTvShow($item) ? $item->tv_title : $item->name;
 
-      if( ! $item) {
-        return null;
+          // See if file is already in our database.
+          if($found = $this->foundInDatabase($title, 'title')) {
+            $this->handleStatus($item, $found->tmdb_id);
+
+            continue;
+          }
+
+          // Otherwise make a new TMDb request.
+          $this->searchTmdb($title, $item);
+        }
+      }
+    }
+
+    /**
+     * Make a new request to TMDb and check against the database.
+     * Otherwise create a new item.
+     *
+     * @param $title
+     * @param $item
+     * @return bool|mixed
+     */
+    private function searchTmdb($title, $item)
+    {
+      $result = $this->tmdb->search($title);
+
+      // If no match found,
+      if( ! $result) {
+        return false;
+      }
+
+      $result = $result[0];
+      $tmdb_id = $result['tmdb_id'];
+
+      // Check against our database.
+      if($this->foundInDatabase($tmdb_id, 'tmdb_id')) {
+        return $this->handleStatus($item, $tmdb_id);
+      }
+
+      // Otherwise create a new item from the result.
+      $created = $this->item->createItem($result, $this->tmdb, $this->storage);
+
+      return $this->handleStatus($item, $created->tmdb_id);
+    }
+
+    /**
+     * See if we can find a item by title or tmdb_id in our database.
+     *
+     * @param $indicator
+     * @return Collection
+     */
+    public function foundInDatabase($value, $type)
+    {
+      if($type == 'title') {
+        return $this->item->searchTitle($value)->first();
+      }
+
+      return $this->item->searchTmdbId($value)->first();
+    }
+
+    /**
+     * Check which status the file has.
+     * Create new src if the status is 'added'.
+     * Update src if status is 'update'.
+     * Remove src if status is 'remove'.
+     *
+     * @param $item
+     * @param $tmdb_id
+     * @return \Exception|mixed
+     */
+    public function handleStatus($item, $tmdb_id)
+    {
+      if($item->status == 'added') {
+        return $this->storeSrc($item, $tmdb_id);
+      }
+
+      if($item->status == 'update') {
+        //
+      }
+
+      if($item->status == 'removed') {
+        //
+      }
+
+      return new \Exception('No status in file found');
+    }
+
+    /**
+     * Handling src store for movie or tv.
+     *
+     * @param $item
+     * @param $tmdb_id
+     * @return mixed
+     */
+    private function storeSrc($item, $tmdb_id)
+    {
+      if($this->isTvShow($item)) {
+        return $this->storeTvSrc($tmdb_id, $item);
       }
-      
-      return $item->tmdb_id;
+
+      return $this->storeMovieSrc($tmdb_id, $item->src);
```

Comment:
Just for the sake of keeping the code unified: maybe use the same params as in storeTvSrc? Instead of **$item->src** use **$item**